src/server.js

```diff
@@ -19,97 +19,175 @@ const timeToClose = 1000 * 60 * 60; // 1 hour in milliseconds
-                lines: rooms.get(roomId).lines,
-                background: rooms.get(roomId).background,
-            });
-            console.log('sent lines');
-        }
-        console.log('rooms: ' + rooms.size);
+        roomId = roomProcess(socket, data);
     });
 
     socket.on('disconnecting', () => {
-        if (rooms.has(roomId)) {
-            rooms.get(roomId).users.delete(socket.id);
-            if (rooms.get(roomId).users.size == 0) {
-                console.log('closing room', roomId);
-                rooms.get(roomId).timeout = setTimeout(
-                    deleteRoom,
-                    timeToClose,
-                    roomId
-                );
-            }
-        }
+        disconnectUser(roomId);
     });
 
     // receiving info about new line from client and sending it to all other clients
     socket.on('draw', (data) => {
-        if (rooms.has(roomId)) {
-            rooms.get(roomId).lines.push(data);
-        }
-        io.to(roomId).emit('draw', data);
+        draw(roomId, data);
     });
 
     socket.on('clearAll', () => {
-        if (rooms.has(roomId)) {
-            rooms.get(roomId).lines = [];
-        }
-        socket.to(roomId).emit('clearAll');
+        clearAllStuff(socket, roomId);
     });
 
     socket.on('background', (data) => {
-        console.log('background changed');
-        if (rooms.has(roomId)) {
-            rooms.get(roomId).background = data;
+        changeBackground(roomId, data);
+        redrawLines(roomId, data);
+    });
+}
+
+function disconnectUser(roomId) {
+    if (rooms.has(roomId)) {
+        rooms.get(roomId).users.delete(socket.id);
+
+        if (rooms.get(roomId).users.size == 0) {
+            ConsoleLog.closingRoom(roomId);
+
+            rooms.get(roomId).timeout = setTimeout(
+                deleteRoom,
+                timeToClose,
+                roomId
+            );
         }
-        io.to(roomId).emit('canvas', {
-            lines: rooms.get(roomId).lines,
-            background: data,
-        });
+    }
+}
+
+function draw(roomId, data) {
+    if (rooms.has(roomId)) {
+        rooms.get(roomId).lines.push(data);
+    }
+
+    io.to(roomId).emit('draw', data);
+}
+
+function clearAllStuff(socket, roomId) {
+    if (rooms.has(roomId)) {
+        rooms.get(roomId).lines = [];
+    }
+
+    socket.to(roomId).emit('clearAll');
+}
+
+function changeBackground(roomId, data) {
+    ConsoleLog.backgroundChange();
+
+    if (rooms.has(roomId)) {
+        rooms.get(roomId).background = data;
+    }
+}
+
+function redrawLines(roomId, data) {
+    io.to(roomId).emit('canvas', {
+        lines: rooms.get(roomId).lines,
+        background: data,
     });
 }
 
 function deleteRoom(roomId) {
     if (rooms.has(roomId)) {
         rooms.delete(roomId);
-        console.log('close room', roomId);
+
+        ConsoleLog.closeRoom(roomId);
+        ConsoleLog.roomsCount();
+    }
+}
+
+function roomExsists(data) {
+    return rooms.has(data);
+}
```

Comment:
You call this function only once and use the entire line from its body in other places. So you should use it everywhere or just remove it.

Reply:
Did not notice. Let me a minute to fix this sh#t